index.bs

```diff
@@ -136,77 +136,166 @@ important to consider ways to mitigate the obvious impacts. For instance:
-mitigations to prevent it from being used to track users across domains or
-without control over clearing this state.  And, are there specific caches
-that a user agent should specially consider?
-
-<p class=example>
-Service Worker [[SERVICE-WORKERS]] intercept all requests made by an
-origin, allowing sites to function perfectly even when offline. A
-maliciously-injected service worker, however, would be devastating (as
-documented in [[SERVICE-WORKERS#security-considerations]]).
-They mitigate the risks an [=active network attacker=] or [=XSS=]
-vulnerability present by requiring an encrypted and authenticated
-connection in order to register a service worker.
-</p>
-
-<p class=example>
-Platform-specific DRM implementations might expose origin-specific
-information in order to help identify users and determine whether they
-ought to be granted access to a specific piece of media. These kinds of
-identifiers should be carefully evaluated to determine how abuse can be
-mitigated; identifiers which a user cannot easily change are very
-valuable from a tracking perspective, and protecting the identifiers from
-an active network attacker is an important concern.
-</p>
+There are many existing mechanisms
+origins can use to
+store information about a user.
+Cookies,
+`ETag`,
+`Last Modified`,
+{{localStorage}},
+and
+{{indexedDB}}
+are just a few examples.
+
+Allowing an origin
+to store data
+on a user’s device
+in a way that persists across browsing sessions
+introduces the risk
+that this state may be used
+to track a user
+without their knowledge or control,
+either in [=first-party-site context|first-=] or [=third-party context|third-party=] contexts.
+
+One of the ways
+user agents mitigate the risk
+that client-side storage mechanisms
+will form a persistent identifier
+is by providing users with the ability
+to clear out the data stored by origins.
+New state persistence mechanisms
+should not be introduced
+without mitigations
+to prevent it
+from being used
+to track users
+across domains
+or without control
+over clearing this state.
+That said,
+manually clearing storage
+is something users do only rarely.
+Spec authors should consider ways
+to make new features more privacy-preserving without full storage clearing,
+such as
+reducing the uniqueness of values,
+rotating values,
+or otherwise making features no more identifying than is needed.
+<!-- https://github.com/w3ctag/design-principles/issues/215 -->
+Also, keep in mind that
+user agents make use of several different caching mechanisms.
+Which, if any, caches will store this new state?
+Are additional mitigations necessary?
+
+<div class=example>
+
+Service Workers
+intercept all requests made by an origin,
+which enables sites
+to continue to function when the browser goes offline.
+Because of this,
+a maliciously-injected service worker
+could be devastating (as documented in [[SERVICE-WORKERS#security-considerations]]).
```

Comment:
Maybe "could compromise the user" rather than "could be devastating"?

Reply:
Fixed in 954fd02.